xbmc/utils/StringUtils.cpp

```diff
@@ -1237,6 +1286,51 @@ int StringUtils::AlphaNumericCollation(int nKey1, const void* pKey1, int nKey2,
   if (r == 0)
     return nKey1 - nKey2;
 
+#ifdef TARGET_ANDROID
+  if (g_langInfo.UseLocaleCollation())
+  {
+    UErrorCode ustatus = U_ZERO_ERROR;
+    try
+    {
+      auto ucoll = std::unique_ptr<UCollator, decltype(&ucol_close)>(
```

Comment:
some of this code is copy-paste from AlphaNumericCompare, so please move the common stuff into smaller functions